### PR TITLE
New asterisk commands

### PR DESCRIPTION
**Pull request by alisinabh**
## Asterisk AGI commands implemented:

- CONTROL STREAM FILE
- GET OPTION
- GET DATA
- RECORD FILE
- WAIT FOR DIGIT
- CHANNEL STATUS

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+language: elixir
```
Comment (marcelog) [inline .travis.yml]: This doesn't belong to this pull, please open a different pull request if needed.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # ElixirAgi
+[![Build Status](https://travis-ci.org/alisinabh/elixir_agi.svg?branch=master)](https://travis-ci.org/alisinabh/elixir_agi)
```
Comment (marcelog) [inline README.md]: This doesn't belong to this pull, please open a different pull request if needed.

File: lib/elixir_agi/agi.ex [hunk 3]
```diff
@@ -189,13 +189,86 @@ defmodule ElixirAgi.Agi do
...
+  end
+
+  #TODO: Implement say_* AST commands
+  #TODO: Implement speeck AST commands
+
+  @doc """
+  Sending commands that are not yet implemeted in elixir_agi is pretty easy
+  you can just use this function with second arg as AST Command name like "STREAM" and third arg should be a list of options for command
+  """
+  @spec custom_cmd(t, String.t, []) :: Result.t
```
Comment (marcelog) [inline lib/elixir_agi/agi.ex]: Question: What's the point of this new function if run/3 is already there?

**Comment by marcelog** [comment]
Hey :)

Thanks for the pull. I've made a few comments on the changes.

Also, could you please squash the commits into 1?

Thanks!

**Comment by alisinabh** [comment]
Hi @marcelog  :)
Thank you very much for your immediate response.

I've squashed commits into 1

but i cant see any comments on code?!

**Review by marcelog — CHANGES_REQUESTED** [review]

**Comment by marcelog** [comment]
Sorry, I forgot to submit the changes. Can you see them now? Thanks!

**Comment by alisinabh** [comment]
Yep, resolved!

sorry i thought run/3 is private :))

**Review by marcelog — APPROVED** [review]

File: lib/elixir_agi/agi.ex [hunk 4]
```diff
@@ -189,13 +189,77 @@ defmodule ElixirAgi.Agi do
...
+  @doc """
+  See: https://wiki.asterisk.org/wiki/display/AST/AGICommand_channel+status
+  """
+  @spec channel_status(t, String.t) :: Result.t
+  def channel_status(agi, channel) do
+    run agi, "CHANNEL STATUS", [channel]
+  end
+
+  #TODO: Implement say_* AST commands
+  #TODO: Implement speeck AST commands
```
Comment (marcelog) [inline lib/elixir_agi/agi.ex]: There are a lot of other ones to implement, perhaps an open issue or pull request is better than these comments in the code

**Review by marcelog — CHANGES_REQUESTED** [review]

**Comment by marcelog** [comment]
Hi @alisinabh!

Thanks for the changes :) I've submitted one more, and when you're done, please squash everything into just 1 commit and I'll merge :)

Best!

**Comment by alisinabh** [comment]
TODO lines removed!
Squashed and ready B)

by the way, i don't know why are we squashing commits? i'm new to github pulls

**Review by marcelog — APPROVED** [review]

**Comment by marcelog** [comment]
Thanks for the contribution @alisinabh!

P.S: Squashing is used in this case to make it look more clean.

Cheers!

**Comment by alisinabh** [comment]
@marcelog thank you very much for your time!

This library is gonna be so useful to me. Looking forward to improve it. 👍 

Great job!

**Comment by marcelog** [comment]
Great! Thanks for your words, looking forward to merge those improvements :)

All the best!

**Pull request merged**